backend/apps/matching/src/matching.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller } from '@nestjs/common';
+import { MessagePattern } from '@nestjs/microservices';
+import { LookingToMatchModel } from './database/models/lookingToMatch.model';
+import { Mutex } from 'async-mutex';
+import { MatchingService } from './matching.service';
+import { MatchingServiceApi } from '@app/microservice/interservice-api/matching';
+
+@Controller()
+export class MatchingController {
+  private mutex = new Mutex();
```

Comment:
This logic should reside in the service layer

Reply:
moved mutex to service layer